### PR TITLE
Fixes #5861: service_ensure_stopped_at_boot is missing

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/5861

### CODE REVIEW

**Comment by amousset** [comment]
WIP

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
@peckpeck Ready for review. Based on #310, itself based on #146.

File: tree/30_generic_methods/service_check_disabled_at_boot.cf [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+# @class_prefix service_check_disabled_at_boot
+# @class_parameter service_name
+# This bundle will define a class service_check_disabled_at_boot_${service_name}_{kept,ok,not_ok,failed,reached}
+
+bundle agent service_check_disabled_at_boot(service_name)
+{
+  vars:
+
+    pass1.systemctl_utility_present::
+      "command_to_check"   string => "${paths.path[systemctl]} is-enabled ${service_name} | grep disabled";
```
Comment (ncharles) [inline tree/30_generic_methods/service_check_disabled_at_boot.cf]: shouldn't the grep be quiet ?
Or even better, exit code of systemctl is 1 for disabled, 0 for enabled

File: tests/acceptance/30_generic_methods/unsafe/service_check_disabled_at_boot.cf [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#######################################################
+#
+# Test if cron is started and if yes, stop it
```
Comment (ncharles) [inline tests/acceptance/30_generic_methods/unsafe/service_check_disabled_at_boot.cf]: Isn't this comment misleading ?
It does ensure that cron is disabled at boot, and check then that it is disabled

Comment (amousset) [inline tests/acceptance/30_generic_methods/unsafe/service_check_disabled_at_boot.cf]: Bad copy/paste!

**Comment by amousset** [comment]
Commit modified

File: tree/30_generic_methods/service_check_disabled_at_boot.cf (hunk 1 above)
Comment (amousset) [inline tree/30_generic_methods/service_check_disabled_at_boot.cf]: The idea it the get an error if the service does not exist. The grep could be quiet but as it is used with returnszero the output is not kept.

Comment (ncharles) [inline tree/30_generic_methods/service_check_disabled_at_boot.cf]: ok great

File: tree/30_generic_methods/service_check_disabled_at_boot.cf [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    pass1.!systemctl_utility_present.chkconfig_utility_present::
+      "command_to_check"   string => "${paths.path[chkconfig]} --list ${service_name} | grep -q 3:off";
```
Comment (peckpeck) [inline tree/30_generic_methods/service_check_disabled_at_boot.cf]: 3 seems to be the runlevel which is not necessarily 3

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
OK, merging this PR

**Pull request merged**